### PR TITLE
[SDK] Fix invalid flow path in run YAML

### PR DESCRIPTION
**Pull request by D-W-**
# Description
This pull request introduces several changes to the `src/promptflow` directory. The most important changes include enhancing the validation of the `flow` property in the `RunSchema` class, adding a new test configuration file for bulk runs, and adding a new test case to validate the behavior of the code when an invalid flow path is provided.

Main changes:

* <a href="diffhunk://#diff-db38cc07d25efdbc0622e8e9352e07e34c502d36bdad1954bcd52db156192c9fL63-R80">`src/promptflow/promptflow/_sdk/schemas/_run.py`</a>: Enhanced the validation of the `flow` property in the `RunSchema` class by adding a new field called `RemoteFlowStr` and a new validation in the `_validate` method. <a href="diffhunk://#diff-db38cc07d25efdbc0622e8e9352e07e34c502d36bdad1954bcd52db156192c9fL63-R80">[1]</a> <a href="diffhunk://#diff-db38cc07d25efdbc0622e8e9352e07e34c502d36bdad1954bcd52db156192c9fR51-R67">[2]</a>
* <a href="diffhunk://#diff-b33d2ea22b9e9679f7a70a7beb5bd27b64c0bdb575e425aeece5322ff550ddbbR1-R11">`src/promptflow/tests/test_configs/runs/bulk_run_invalid_flow.yaml`</a>: Added a new test configuration file for bulk runs, `bulk_run_invalid_flow.yaml`.
* <a href="diffhunk://#diff-c3d1c4e4539af1a59525218043dad93dc866b761a70a16a21783e57a7d0adac5R97-R103">`src/promptflow/tests/sdk_cli_test/unittests/test_run.py`</a>: Added a new test case to validate the behavior of the code when an invalid flow path is provided.

Other changes:

* <a href="diffhunk://#diff-41ec3f7c4b5d4c0e670407d3c00a03a6966d7ebf617b1536473e33a12e2bc765R5-R12">`src/promptflow/CHANGELOG.md`</a>: Introduced a new feature to the Executor, a `@trace` decorator, which allows logging traces for functions called by tools. However, it was later decided to remove this decorator without mentioning the reason in the diff. <a href="diffhunk://#diff-41ec3f7c4b5d4c0e670407d3c00a03a6966d7ebf617b1536473e33a12e2bc765R5-R12">[1]</a> <a href="diffhunk://#diff-41ec3f7c4b5d4c0e670407d3c00a03a6966d7ebf617b1536473e33a12e2bc765L18-R23">[2]</a>

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,14 @@
 
 ## 1.4.0 (Upcoming)
 
+### Features Added
+
+- [Executor] Add @trace decorator to make it possible to log traces for functions that are called by tools.
```
Comment (D-W-) [inline src/promptflow/CHANGELOG.md:7]: @zzn2 please help check if this is as expected.

File: src/promptflow/promptflow/_sdk/schemas/_run.py [hunk 2]
```diff
@@ -48,6 +48,23 @@ def _validate(self, value):
             )
 
 
+class RemoteFlowStr(fields.Str):
```
Comment (D-W-) [inline src/promptflow/promptflow/_sdk/schemas/_run.py:51]: @0mza987 please help check if this works as expected

Comment (0mza987) [reply src/promptflow/promptflow/_sdk/schemas/_run.py:51]: In error message, "azureml:xxx" -> "azureml:<flow-name>"? And there seems to be a typo here:
image

File: src/promptflow/tests/sdk_cli_test/unittests/test_run.py [hunk 3]
```diff
@@ -94,6 +94,13 @@ def test_dot_env_resolve(self):
         run = load_run(source=source, params_override=[{"name": run_id}])
         assert run.environment_variables == {"FOO": "BAR"}
 
+    def test_run_invalid_flow_path(self):
```
Comment (0mza987) [inline src/promptflow/tests/sdk_cli_test/unittests/test_run.py:97]: Also add a test to guard invalid remote flow error?

Comment (D-W-) [reply src/promptflow/tests/sdk_cli_test/unittests/test_run.py:97]: Sure, let me add one

File: src/promptflow/promptflow/_sdk/schemas/_run.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/schemas/_run.py:51]: Updated

**Review by 0mza987 — APPROVED** [review]

**Review by guming-learning — APPROVED** [review]

File: src/promptflow/CHANGELOG.md (hunk 1 above)
Comment (zzn2) [reply src/promptflow/CHANGELOG.md:9]: Looks good to me.

**Pull request merged**